vncviewer/Win32AudioOutput.cxx

```diff
@@ -0,0 +1,303 @@
+/* Copyright 2022 Mikhail Kupchik
+ *
+ * This is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this software; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307,
+ * USA.
+ */
+
+#include "Win32AudioOutput.h"
+
+Win32AudioOutput::Win32AudioOutput()
+  : haveWO(false), openedWO(false), sampleFormat(sampleFormatU8), numberOfChannels(0),
+    samplingFreq(0), currentStreamId(0), handleWO(NULL), bufPtr(NULL), bufTotalSize(0),
+    bufFreeSize(0), bufUnsubmittedSize(0), bufSubmittedHead(0), bufUnsubmittedHead(0),
+    doneHdrsSlist(NULL), hdrsInFlight(0), extraDelayInMillisec(0)
+{
+  static const rdr::U32 freqTable[4] = {44100, 48000, 22050, 11025};
+
+  for (rdr::U8 bits_per_sample = 16; bits_per_sample != 0; bits_per_sample -= 8) {
+    for (rdr::U8 n_channels = 2; n_channels != 0; n_channels--) {
+      for (rdr::U8 freq_index = 0; freq_index < 4; freq_index++) {
```

Comment:
Doesn't Windows resample for us? Do we really need to try all of these? It would make things a lot simpler if we just ha a fixed format we always use.

Reply:
All of this adds a lot of complexity to the code. So if it isn't needed in practice, then it's just a cost.

Have you actually seen it failing to give you the requested format?